src/frontends/common/shutdown_protobuf/shutdown_protobuf.cpp

```diff
@@ -0,0 +1,26 @@
+#include <google/protobuf/stubs/common.h>
+
+#if defined(_WIN32)
+#    include <windows.h>
+BOOL WINAPI DllMain(HINSTANCE hinstDLL,  // handle to DLL module
+                    DWORD fdwReason,     // reason for calling function
+                    LPVOID lpReserved)   // reserved
+{
+    // Perform actions based on the reason for calling.
+    switch (fdwReason) {
+    case DLL_PROCESS_ATTACH:
+    case DLL_THREAD_ATTACH:
+    case DLL_THREAD_DETACH:
+        break;
+
+    case DLL_PROCESS_DETACH:
+        google::protobuf::ShutdownProtobufLibrary();
+        break;
+    }
+    return TRUE;  // Successful DLL_PROCESS_ATTACH.
+}
+#elif defined(__linux__)
```

Comment:
Do we need to have something like this for macOS?

Reply:
@ilyachur have updated the code for macOS